web/src/engine/main/src/keymanEngine.ts

```diff
@@ -54,12 +54,7 @@ export default class KeymanEngine<
       }
     }
 
-    //... probably only applies for physical keystrokes.
-    if(!event.isSynthetic) {
-      if(this.osk?.vkbd?.keyPending) {
-        this.osk.vkbd.keyPending = null;
-      }
-    } else if(this.keyEventRefocus) { // && event.isSynthetic // as in, is from the OSK.
+    if(this.keyEventRefocus) { // && event.isSynthetic // as in, is from the OSK.
```

Comment:
Given this is no longer an else from `!event.isSynthetic`:

```suggestion
    if(this.keyEventRefocus) {
```

or

```suggestion
    if(this.keyEventRefocus && event.isSynthetic /* as in, is from the OSK. */) {
```